Rename update

sisou opened the pull request:
Two issues remaining with this PR:

1. ~~I cannot get the account label to be limited to the page width. It always grows too large when the input value becomes longer.~~
2. For the new input styles, I have to use @nimiq/style v0.6.0. In v0.5.0, the way icons are handled has changed. To not have to update the whole Accounts Manager, I just included the v0.4.1 icons CSS file for now.

Actual screenshot of implementation:

![image](https://user-images.githubusercontent.com/1828163/56264688-91b85780-60e7-11e9-842b-8dc464122778.png)

Bettelstab commented:
> Is the visibility toggle left out on purpose?

I'd say another PR for that is fine.

Bettelstab commented:
> I think the problem with the width lies in vue-components `LabelInput`. The updateWidth function should not take `Math.max`, but remove any set width in case `placeholderWidth < valueWidth`, leaving it to css then. Maybe we can even get rid of that part completely.

@sisou  Just experimented with CSS, you CAN get it right with a lot of max-widths, but probably @nibhar's appproach is better :D

sisou commented:
I fixed the width and centering issue and also added the default labels for addresses and the account as the placeholder and fallbacks, when users leave the in put empty.

danimoh commented on src/views/Rename.vue:
```diff
@@ -174,15 +180,47 @@ export default class Rename extends Vue {
         padding: 2rem 1rem;
     }
 
+    :global(.label-input) {
```
What does `:global` do? Is it like using `>>>`?

sisou replied:
Similar. But I honestly forgot about `>>>`. I have refactored the styles to use `>>>` instead.

sisou commented on src/views/Rename.vue:
```diff
@@ -15,10 +15,11 @@
             </PageFooter>
             <transition name='fade-in'>
                 <Loader v-if="labelsStored"
+                    :lightBlue="true"
```
For consistency with the other attributes, I like making it explicit.

danimoh replied:
What's this actually needed for?
Due to `state="Success"` it's green anyways.

I'm merging right now.
Is this line safe to delete?

sisou replied:
It should be used so that the loader transitions from light-blue to green during the animation.

danimoh replied:
However as the state is already `"Success"` when the component renders for the first time, it does not transition.
I'm thus removing this line.

sisou commented on src/views/Rename.vue:
```diff
@@ -104,16 +89,37 @@ export default class Rename extends Vue {
+            ...this.wallet.accounts.values(),
+            ...this.wallet.contracts,
+        ].map((accountOrContract) => {
+            accountOrContract.balance = undefined; // Hide balances in list, as they may be terribly out-of-date
```
Solved by hiding the amount with CSS from the Rename view.

sisou commented on src/views/Rename.vue:
```diff
@@ -104,16 +89,37 @@ export default class Rename extends Vue {
+            ...this.wallet.contracts,
+        ].map((accountOrContract) => {
+            accountOrContract.balance = undefined; // Hide balances in list, as they may be terribly out-of-date
+            accountOrContract.placeholder = LabelingMachine.labelAddress(accountOrContract.userFriendlyAddress);
```
I like the `defaultLabel` accessor. Will refactor.

danimoh commented on src/lib/AccountInfo.ts:
```diff
@@ -19,6 +20,7 @@ export class AccountInfo {
     }
 
     public walletId?: string;
+    public placeholder?: string;
```
I think these are not needed anymore.
